### PR TITLE
Add Anymal-D torchrl cfg

### PR DESCRIPTION
**Pull request by fyu-bdai**
# Description

Adds TorchRL training configuration for the Anymal-D velocity environment as a template for training IsaacLab environments with the new torchrl training workflow. 

You can try training Anymal with TorchRL (after merging all TorchRL PRs) using 
`/workspace/isaaclab/source/standalone/workflows/torchrl python train.py --task Isaac-Velocity-Flat-Anymal-D-v0 --num_envs 4096`

Related PRs:
#1178, #1179 

This is the last PR in the group of 3 that adds the TorchRL training pipeline. 

## :warning: The not so good parts :warning:
Unfortunately, the Anymal-D environment converges slower than RSL-RL. This is probably due to policy architecture and PPO implementation differences which requires different hyperparameter settings. I have provided the best hyperparameters that have worked for me so far. While it is possible to speed up the convergence by increasing `desired_kl` targets and `entropy_coef` to closely match RSL-RL, torchrl policies seem to crash late during training due to spurious KL/action noise spikes after the reward has long converged. 

## Training curves and video

Screenshot from 2024-10-08 00-22-03

Screenshot from 2024-10-08 00-22-16

https://github.com/user-attachments/assets/f946e82f-2ef0-4a1e-9cfd-1ff7f45903b5

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/manager_based/locomotion/velocity/config/anymal_d/__init__.py [hunk 1]
```diff
@@ -51,6 +54,6 @@
     kwargs={
         "env_cfg_entry_point": rough_env_cfg.AnymalDRoughEnvCfg_PLAY,
         "rsl_rl_cfg_entry_point": f"{agents.__name__}.rsl_rl_ppo_cfg:AnymalDRoughPPORunnerCfg",
-        "skrl_cfg_entry_point": f"{agents.__name__}:skrl_rough_ppo_cfg.yaml",
```
Comment (Toni-SM) [inline source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/manager_based/locomotion/velocity/config/anymal_d/__init__.py]: Why `skrl` config was replaced by the `torchrl` one?

Comment (fyu-bdai) [reply source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/manager_based/locomotion/velocity/config/anymal_d/__init__.py]: Copy paste error, sorry!

File: source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/manager_based/locomotion/velocity/config/anymal_d/agents/torchrl_ppo_cfg.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+from omni.isaac.lab_tasks.utils.wrappers.torchrl.torchrl_ppo_runner_cfg import (
+    ClipPPOLossCfg,
+    CollectorCfg,
+    OnPolicyPPORunnerCfg,
+    ProbabilisticActorCfg,
+    ValueOperatorCfg,
+)
+
+
+class AnymalDActorNN(nn.Module):
```
Comment (Toni-SM) [inline source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/manager_based/locomotion/velocity/config/anymal_d/agents/torchrl_ppo_cfg.py:20]: Question: How having the model manually defined in the agent config will support changing it from CLI (e.g.: using hydra)?

Comment (fyu-bdai) [reply source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/manager_based/locomotion/velocity/config/anymal_d/agents/torchrl_ppo_cfg.py:20]: I suppose we can add an argument to the torchrl CLI args to specify which model definition method the user would like to use if we want to support both methods?

**Comment by AntoineRichard** [comment]
Hi @fyu-bdai — thank you for opening this PR, and for the effort you put into it! 🙏

We're cleaning up the Isaac Lab pull request backlog. **Why this PR is being closed:** This is a duplicate torchrl-workflow effort with #1179. Here is exactly what we found on this PR when we reviewed the backlog:

| | |
|---|---|
| **Opened** | 2024-10-08 (over a year ago) |
| **Last commit on the branch** | 2024-10-08 |
| **Last activity from the author** | over a year ago |
| **Target branch** | `main` — note that active development happens on `develop`; `main` only receives releases |
| **Review status** | Reviewed by Toni-SM, but it was never brought to a conclusion |
| **Merge status** | Conflicts with the base branch — it no longer applies cleanly |
| **Size** | 5 commit(s), 3 file(s) changed, +201 / -0 |

It was picked up by the sweep because it targets `main` instead of `develop`, and it has been open for over a year. It was then put in the "close" bucket because the author has been silent for over a year — which is the signal we used to tell apart pull requests that are still being worked on from ones that have genuinely been set aside.

We deliberately did **not** close pull requests that were approved and ready to land, or that were small and clearly still fixing a live bug — there were 27 of those, and we are merging them rather than closing them.

We'd much rather see this land as a single consolidated PR than have two parallel efforts both go stale — so this really isn't a rejection of the work.

**One thing worth knowing first:** Isaac Lab 3.0 has since shipped. `develop` is now at
3.0.0 while `main` is still at 2.3.2, and 3.0 restructured a fair amount — quaternion
conventions, actuator collections, data access, and the Isaac Sim extension imports all
changed. The [migration guide](https://github.com/isaac-sim/IsaacLab/blob/develop/docs/source/migration/migrating_to_isaaclab_3-0.rst)
is the place to start if you rebase.

It is also the honest reason a lot of these older pull requests stopped applying: the code
they touched moved or was rewritten underneath them.

**If you'd like to carry it forward, please reopen this PR or open a fresh one against `develop`** and we'll help get the two efforts merged into one. 💚

---
🤖 *This comment was drafted with AI assistance as part of a maintainer-led sweep of the Isaac Lab pull request backlog. A maintainer is behind this cleanup — but if this closure looks wrong, it may well be, so please push back and we'll take another look.*
